doc/source/download.rst

```diff
@@ -10,6 +10,160 @@ Download
        :depth: 3
        :backlinks: none
 
+Binaries
+------------------------------------------------------------------------------
+
+The GDAL project itself does not produce regular downloadable binaries
+(executables) for each release. However, a variety of efforts do produce
+ready to use binaries.
+
+Note that the maintainers of those distributions are generally not the maintainers
+of the GDAL sources, so please report any issue specific to those builds through
+their own support channels.
+
+Windows, macOS and Linux
+................................................................................
+
+See **https://trac.osgeo.org/gdal/wiki/DownloadingGdalBinaries** for a list
```

Comment:
I don" think we don't want to link to Trac wiki, because most content there is outdated, and the general trend is to move content from Trac wiki to here. Parts that are still relevant should be imported into this repository under doc/source.

Reply:
@jmckenna what do you think of that?